script/binary

```diff
@@ -17,7 +17,7 @@
 set -e
 
 if [ -z "$1" ]; then
```

Comment:
Sure - there can be several bash-foo improvements here, not embarking on those to avoid losing this PR focus (i.e. just really fixing that buggy usage)